mailpoet/lib/Util/Security.php

```diff
@@ -31,6 +32,44 @@ public static function generateToken($action = 'mailpoet_token') {
     return WPFunctions::get()->wpCreateNonce($action);
   }
 
+  public static function generateTokenForUniqueAction(string $action) {
```

Comment:
Not sure why we call it `generateToken` when internally we call `wpCreateNonce` (what I mean is we could use the "nonce" naming also outside to be clearer).

But I'm also not sure why the method was in this file. Maybe we could extract it now to a dedicated class to handle the nonces only?

Reply:
Maybe we should just use the `wpCreateNonce` and `wpVerifyNonce` methods from `WPFunctions` directly wherever we need them? It doesn't seem like we need any special functionality unless we want to hardcode the action name to always be `mailpoet_token`, which doesn't seem like a great idea anyway.

Reply:
> Maybe we should just use the wpCreateNonce and wpVerifyNonce methods from WPFunctions directly wherever we need them?

True, also a possibility. I wrote that comment before thinking about the caching and validity of the ticket itself.